subsys/bluetooth/host/adv.c

```diff
@@ -2064,8 +2064,12 @@ void bt_hci_le_per_adv_response_report(struct net_buf *buf)
 		response = net_buf_pull_mem(buf, sizeof(struct bt_hci_evt_le_per_adv_response));
 		info.tx_power = response->tx_power;
 		info.rssi = response->rssi;
-		info.cte_type = BIT(response->cte_type);
 		info.response_slot = response->response_slot;
+		if (response->cte_type == BT_HCI_LE_NO_CTE) {
+			info.cte_type = BT_DF_CTE_TYPE_NONE;
+		} else {
+			info.cte_type = BIT(response->cte_type);
```

Comment:
This assumes that the `BT_DF_CTE_TYPE_*` values are defined in a specific way. We generally do not make such assumptions which is why we have functions like

```c
static uint8_t get_hci_cte_type(enum bt_df_cte_type type)
{
	switch (type) {
	case BT_DF_CTE_TYPE_AOA:
		return BT_HCI_LE_AOA_CTE;
	case BT_DF_CTE_TYPE_AOD_1US:
		return BT_HCI_LE_AOD_CTE_1US;
	case BT_DF_CTE_TYPE_AOD_2US:
		return BT_HCI_LE_AOD_CTE_2US;
	default:
		LOG_ERR("Wrong CTE type");
		return BT_HCI_LE_NO_CTE;
	}
}
```

Suggest to create a function that converts HCI types to `BT_DF_CTE_TYPE_*` to make it future proof.

Reply:
I think it will require adding this function to direction.c/direction.h and adding direction.h as a dependency to adv.c.
I believe it is a redundant dependency, so I would suggest to keep the implementation as is for now

Reply:
It does not need to be a public function

Reply:
I added an inline function to direction.h
Please let me know if you would prefer other implementation